src/chart_types/partition_chart/layout/viewmodel/viewmodel.ts

```diff
@@ -59,11 +58,29 @@ import {
 } from '../utils/group_by_rollup';
 import { StrokeStyle, ValueFormatter } from '../../../../utils/commons';
 import { percentValueGetter } from '../config/config';
+import { $Values } from 'utility-types';
+
+// todo consider turning it into a config option
+const topGroove = 20;
 
 function paddingAccessor(n: ArrayEntry) {
   return entryValue(n).depth > 1 ? 1 : [0, 2][entryValue(n).depth];
 }
-function rectangleFillOrigins(n: ShapeTreeNode): [number, number] {
+
+function topPaddingAccessor(topGroovePx: Pixels) {
+  return (n: ArrayEntry) => (entryValue(n).depth === 1 ? topGroovePx : paddingAccessor(n));
+}
+
+export const VerticalAlignments = Object.freeze({
+  top: 'top' as 'top',
+  middle: 'middle' as 'middle',
+  bottom: 'bottom' as 'bottom',
+});
+
+/** @internal */
+export type VerticalAlignment = CanvasTextBaseline & $Values<typeof VerticalAlignments>;
```

Comment:
This kind of combined type is confusing. I think these should be separate types. This implies that the value `VerticalAlignments` has the same types as the `VerticalAlignments` which is not the case.

Fro example `VerticalAlignments` object does not have `'alphabetic'` as a possible value.

```suggestion
export type VerticalAlignment = $Values<typeof VerticalAlignments>;
export type AllVerticalAlignment = CanvasTextBaseline & VerticalAlignment;
```

Reply:
It can certainly be broken up eg.
```
type PermittedVerticalAlignments = $Values<typeof VerticalAlignments>;
export type VerticalAlignment = CanvasTextBaseline & PermittedVerticalAlignments;
```
which just lifts and names the `$Values` part. Would that be clearer code? Your suggestion is the same extraction but I don't follow the namings as `AllVerticalAlignments` seems to hint that it's a broader / union set, while the purpose of the `&` is to intersect the two sets. Also, both do not need to be exported, but maybe you have some non-local change in mind.

OK why the heck did I do this in the first place, ie. why not just the expected `$Values<typeof VerticalAlignments>`? The `&`-ing will not in fact remove values from `$Values<typeof VerticalAlignments>` (one is a proper subset of the other). The purpose is to show intent and assurance to the compiler and code reader that 
- values are expected to be directly digestible by the Canvas2 API without compile time type juggling or worse, runtime type guards, as it's directly channeled into a `ctx.textBaseline` so it better be of `CanvasTextBaseline`(*)
- yet, as you say, we don't currently expose all the alignments

There is a larger, somewhat theme and it'd be great to talk about it on account of this, we touched on esp. text styling configurability superficially with @markov00 and it'd be opportune to use this as a small example.

Reply:
broken up, as mentioned, in a slightly different way in https://github.com/elastic/elastic-charts/pull/652/commits/cb141a6402abf7918c8e54e8e2803d5c872f66dd

Reply:
Yeah the `AllVerticalAlignments` was just an arbitrary name I cam up with. 

The change in cb141a6 is close to what I was referring to but the key is that the object with the js values and the type should be called the same, that way ts can import the Type and Values as one import.

```ts
// also using this
type SupportedVerticalAlignments = $Values<typeof VerticalAlignments>;

// makes this look strange
const myThing: SupportedVerticalAlignments = VerticalAlignments.top
```

These values are basically the poor mans `enum` because tsc enums are not ideally transpiled (see https://github.com/elastic/kibana/pull/62957#discussion_r406385788), with this thinking, your enum values in this case would be different than the enum type.

```ts
export const VerticalAlignments = Object.freeze({
  top: 'top' as 'top',
  middle: 'middle' as 'middle',	
  bottom: 'bottom' as 'bottom',
});
type VerticalAlignments = $Values<typeof VerticalAlignments>;

// then it becomes much clearer how the type and enum values relate
type VerticalAlignments = $Values<typeof VerticalAlignments>;
```

Could be something to talk about today at the sync 🤷

Reply:
Yes, the enum would have various weaknesses here, one of which here is that it's not some opaque thing I want here, but a subtype of `CanvasTextBaseline` which is a very real set of strings.

If I understand, your remaining ask is that the object and type be named identically; it's doable, though with a bit less directness here as there's the above mentioned goal of piggybacking on the `CanvasTextBaseline` type, in https://github.com/elastic/elastic-charts/pull/652/commits/9c300904c6cdedf486265ffe042742b4779bafd9